pkg/ddl/job_worker.go

```diff
@@ -970,7 +973,12 @@ func toTError(err error) *terror.Error {
 
 // updateGlobalVersionAndWaitSynced update global schema version to notify all TiDBs
 // to reload info schema, and waits for all servers' schema or MDL synced.
-func updateGlobalVersionAndWaitSynced(jobCtx *jobContext, latestSchemaVersion int64, job *model.Job) error {
+func updateGlobalVersionAndWaitSynced(
+	ctx context.Context,
+	jobCtx *jobContext,
+	latestSchemaVersion int64,
+	job *model.Job,
+) error {
```

Comment:
line 993 is using jobCtx.ctx, it might now working if state overwritten, see https://github.com/pingcap/tidb/issues/52747
- job worker change to done, but before we wait cancelled by other session
- cancel check routine cancelled it
- we start wait version using a cancelled context, never work

Reply:
https://github.com/pingcap/tidb/pull/56404/commits/41a7af827efedc60521e5c435787c471005d1e39 not sure I understand it correctly.